submissions/ORybka/dom-js/main.js

```diff
@@ -0,0 +1,222 @@
+    buttonLi = document.createElement('li');
+    button = document.createElement('button');
+    button.className = 'menu-item';
+    button.id = id;
+    button.innerHTML = `
+    <img class="menu-item-img" src="assets/shields/${name}.png"><span>house ${name}</span>`;
+    buttonLi.appendChild(button);
+    menu.appendChild(buttonLi);
+  });
+}
+
+function changeContent(e) {
+  let target = e.target;
+  if (!target.matches('button')) {
+    target = target.parentNode;
+  }
+  const btn = target.closest('.menu-item');
+  if (btn) {
+    id = target.getAttribute('id');
+    createElements(houses[id]);
+    addImages(houses[id]);
+    animateMap(houses[id]);
+  }
+}
+
+menu.addEventListener('click', changeContent);
+
+function createElements({ name, slogan, description }) {
+  houseElement.innerText = `house ${name}`;
+  shieldElement.setAttribute('src', `assets/shields/${name}.png`);
+  sloganElement.innerText = `${slogan}`;
+  descriptionElement.innerHTML = `<span>${description}</span>`;
+  background.style.backgroundColor = `var(--${name})`;
+}
+
+function addImages(item) {
+  if (images.children.length === 0) {
+    createImage(item);
+  } else {
+    images.innerHTML = '';
+    createImage(item);
+  }
+}
+
+function createImage({ name }) {
+  for (let i = 0; i < number; i++) {
+    image = document.createElement('div');
+    image.className = 'content-img';
+    image.id = i;
+    images.appendChild(image);
+    image.style.backgroundImage = `url("assets/members/member-${name}-${i}.jpg")`;
+    if (imageCollection.length >= 3) {
+      imageCollection = [];
+    }
+    imageCollection.push(image);
+  }
+}
+
+function animateMap(item) {
+  document.querySelector('.background-picture').animate(
+    [
+      { transform: `${houses[previousId].mapPosition[0]}`, bottom: `${houses[previousId].mapPosition[1]}`, right: `${houses[previousId].mapPosition[2]}` },
+      { transform: `${item.mapPosition[0]}`, bottom: `${item.mapPosition[1]}`, right: `${item.mapPosition[2]}` },
+    ],
+    {
+      duration: 1000,
+      fill: 'forwards',
+    }
+  );
+  previousId = item.id;
+}
+
+function changeImageBackground(img, i) {
+  if (!img.classList.contains('fire') && !img.classList.contains('light') && !img.classList.contains('dead')) {
+    if (houses[id].dead[i] === 1) {
+      img.style.backgroundImage = '';
+      img.classList.add('fire');
+      showDead(img);
+    } else {
+      img.style.backgroundImage = '';
+      img.classList.add('light');
+    }
+
+    setTimeout(() => {
+      img.style.backgroundImage = `url("assets/members/member-${houses[id].name}-${i}.jpg")`;
+    }, time);
+  }
+}
+
+function showDead(el) {
+  setTimeout(() => {
+    const deadImage = document.createElement('div');
+    deadImage.className = 'dead';
+    el.appendChild(deadImage);
+  }, time);
+}
+
+function addImageAnimation(e) {
+  const target = e.target;
+  const img = target.closest('.content-img');
+  imageId = target.getAttribute('id');
```

Comment:
why do use store this variable in a global scope? You can easily use const here and use it as a local variable. No need to store it as a global, because it can be hard to find it and you can accidentaly reassign it in some functions. It's not a safe approach

Reply:
You are totally right! Everything works great with local variables! That was quite strange mistakes :(